### ISSUE TITLE
Question: Should zend.assertions = 1 work?

### ISSUE THREAD

**Issue by DanielSiepmann** [phpactor/phpactor#1610]
I've modified my setup resulting in `zend.assertions = 1` which breaks the setup. Calling the context menu results in:
```
Error detected while processing function phpactor#ContextMenu[1]..phpactor#rpc:
line   20:
E605: Exception not caught: Could not parse response from Phpactor: Vim(let):E474: Unidentified byte: Warning: assert(): assert($node instanceof AssignmentExpression) failed in /nix/store/im1r2bkl
9i73g0862w8s1ajpdd6h6sn9-vimplugin-phpactor-042c2f2733189118f8fe95a03b9701e43c81c98f/lib/WorseReflection/Core/Inference/Walker/PassThroughWalker.php on line 40^@Warning: assert(): assert($node ins
tanceof AssignmentExpression) failed in /nix/store/im1r2bkl9i73g0862w8s1ajpdd6h6sn9-vimplugin-phpactor-042c2f2733189118f8fe95a03b9701e43c81c98f/lib/WorseReflection/Core/Inference/Walker/PassThroug
hWalker.php on line 40^@Warning: assert(): assert($node instanceof AssignmentExpression) failed in /nix/store/im1r2bkl9i73g0862w8s1ajpdd6h6sn9-vimplugin-phpactor-042c2f2733189118f8fe95a03b9701e43c
81c98f/lib/WorseReflection/Core/Inference/Walker/PassThroughWalker.php on line 40^@Warning: assert(): assert($node instanceof AssignmentExpression) failed in /nix/store/im1r2bkl9i73g0862w8s1ajpdd6
h6sn9-vimplugin-phpactor-042c2f2733189118f8fe95
```

My old setup had `zend.assertions = -1` and reverting to `-1` removes the issue.

That's why I would kindly ask whether `zend.assertions = 1` should work?
https://www.php.net/manual/en/function.assert.php#function.assert.expectations documents the behaviour:

> 1: generate and execute code (development mode)
> 0: generate code but jump around it at runtime
> -1: do not generate code (production mode)

**Comment by dantleech** [comment]
hm, i think it should. that is new code and is clearly wrong (the `assert` can be removed). should change to run the tests with zend assertions = 1

**Comment by DanielSiepmann** [comment]
Thanks :) Also first time I encountered that, but I'm still not using `assert()` myself.

I keep the setup for now as it is working. Will update my setup once you close this issue.

**Comment by dantleech** [comment]
fixed in #1613 and also uncovered an issue or two :+1:

**Comment by DanielSiepmann** [comment]
Saw your PR, thanks :)
Already updated and it works without changing the setting. Thank you.
